pkg/http/header/header.go

```diff
@@ -126,8 +128,7 @@ func IsProbe(r *http.Request) bool {
 
 // IsKubeletProbe returns true if the request is a Kubernetes probe.
 func IsKubeletProbe(r *http.Request) bool {
-	return strings.HasPrefix(r.Header.Get("User-Agent"), KubeProbeUAPrefix) ||
-		r.Header.Get(KubeletProbeKey) != ""
+	return strings.HasPrefix(r.Header.Get("User-Agent"), KubeProbeUAPrefix)
```

Comment:
I think we should leave this as is for now and then remove the use of `K-Kubelet-Probe` after a release or two

Reply:
if there's a minor drift between the queue proxy and the deployment then we want to prevent things from blowing up.